### DISCUSSION TITLE
W3C Standards for Linked Data and Semantic Web for a Large Enterprise

### DISCUSSION THREAD

**Discussion by jimschoening** [CommonCoreOntology/CommonCoreOntologies#326] [Announcements]
This topic questions whether compliance with World Wide Web Consortium (W3C) standards for Linked Data and Semantic Web can enable IC/DOD-wide capabilities to aggregate knowledge derived from disparate data in many domains and 1000's of siloed tools, or can Linked Data, if its many systems chose to voluntarily adopt a standard top-level ontology and mid-level ontology, and an enterprise ontology foundry that ensures domain extension ontologies d0 not overlap, achieve the same aggregated knowledge outcome?

**Comment by J-Iannazzo** [comment]
Thank you Jim.  Recommend adjusting the topic question to:
This topic questions whether compliance with World Wide Web Consortium (W3C) standards for Linked Data and Semantic Web can enable IC- and DOD-wide capabilities to aggregate knowledge derived from disparate data in many domains and 1000's of siloed tools, or can Linked Data defined by (Jim, please cite the standard reference other than W3C) work to achieve the same aggregated knowledge outcome?

**Comment by jimschoening** [comment]
I'll kick this off.  Linked Data achieves data interoperability by linking data elements from one model to another.  This does work, but the big question is, can it scale to a large diverse (i.e. many domains) enterprise? Doesn't it face the classic N-Squared problem?

**Comment by J-Iannazzo** [comment]
If the Internet is understood to be a large diverse enterprise by the members of this forum, then absolutely yes, W3C-compliant Linked Data and Semantic Web capability is a proven capability.  Facebook, Netflix, Apple, Google, Amazon, Linked In, Uber, Uber Eats, You Tube, and many more commercial sectors have applied and benefited from the Linked Data standards prescribed by the W3C.  To understand Linked Data principles and practices in this context, the forum must be knowledgeable of W3C standard for Linked Data.  This standard requires diligence towards assigning meaning to data (Ontologies), identifying contents and associated Ontology instantiations applied to the contents, and conveyance of the collective knowledge derived from the identifier attached to the contents and associated Ontology instantiations.  When applied in this fashion, an enterprise distinguishes contributing systems and data therein from aggregating systems and the knowledge therein.  An aggregating system is referred to as an Enterprise Knowledge Graph; to the hundreds of members comprising the Cogent Way Community of Interest, there is no Enterprise Knowledge Graph system in existence in the IC and DOD to-date.  Consequently, contributing systems cause the Tool-centric vice Data-centric approach evident in today's modernization activities (e.g., ADVANA, PALANTIR, MARS, and more).   When the W3C-compliant Enterprise Knowledge Graph system (system-of-systems) enters into the IC and DOD enterprise, then contributing systems must shift to condition their data to be compliance with W3C standards for Linked Data and Semantic Web.  A reference that describes Linked Data in W3C context and the opportunity to introduce the many changes that would be required to employ these Linked Data Principles is attached.  More information to share and this level-setting serves to provide context.

**Comment by J-Iannazzo** [comment]
Trying to attach the file referenced in my comment above.

**Comment by VictorRohr** [comment]
Concur with Jeff that there is no "Enterprise Knowledge Graph system" in place today.  More importantly though, is to not think of this knowledge graph as a system.  Rather, it is the means by which shards of information can be related across a vast enterprise - and this is what is enabled by the W3C rdf recommendation.  If enterprise data can be explicitly referenced by URIs or IRIs, then the handles exist allowing for this data to be correlated using rdf triples.  The other side of this is the need for explicit, unambiguous expressivity of information.  The latter is achieved using ontologies where all classes, properties, and individuals in the ontology can also be explicitly referenced by URI/IRI.  In so doing, the meaning of things is globally grounded.  The next concern is that this suite of ontologies that is selected for use, is one that is constructed to assure continued future use and limitless extensibility.  This requirement suggests the adoption of a top-level ontology approach which has a proven 15+ year track record of success and is now codified in requirements specified in ISO/IEC 21838-1.  By adopting BFO as the top level ontology and CCO as the mid-level ontology, not only does an enterprise position itself for future use assuring data interoperability and interpretability, but  also reduces the burden on developers through ontology reuse.

**Comment by alanruttenberg** [comment]
Could the issue be clarified? Is there an opposition here, alternatives that we are trying to decide between? Are there CCO development choices that hinge on the outcome of this issue? Questions about how CCO would be used? The impetus in the development of linked data was precisely to enable wide-scale integration of data from many sources, so the answer to the question as asked would be "yes" - at least that's the intent. Does anyone disagree? There are various kinds of comments I could make, but I'd like to be responsive to what the concerns are.

**Comment by jimschoening** [comment]
I think the issue is whether 'W3C Standards for Linked Data and Semantic Web' can be implemented within a large and diverse enterprise (like DOD/IC) without a hub ontology.  I believe they cannot, but Jeff might be asserting they can, so he should make his case.  But I believe he is also suggesting the use of DICO, so maybe he is proposing a hub ontology, and if he is, maybe it doesn't matter if he believes a hub ontology is needed or not.  We'll see his position next week.

**Comment by J-Iannazzo** [comment]
Happy Monday!  I hope that everyone had a great Holiday weekend.  Alan, I do not think there is opposition.  We have been consistent in a position that focuses on the viability of W3C Standards for Linked Data and Semantic Web.  An integral expectation of W3C is the use of Ontology in conjunction with interdependent principles such as Uniform Resource Identifiers.  In our work with a broad Community of Interest including DIA ADO-3 who are core Partners with us and championing DICO, there is no Enterprise Knowledge Graph capability evident in the IC and DOD that we know about.  As we discuss this topic with CDO Partners and AI/ML Offices of Primary Responsibility, we are introducing for their consideration W3C standards/practices to consider as a way to shift from tool-centric to data-centric investments.  We believe that W3C Standards should be considered by the CDO and CIO decision makers, and with these standards can come options to achieve Linked Data expectations.  Options as we have risk-reduced in prototyping efforts in the past and are seeking to further risk-reduce in R&D efforts going forward include a distinction between data and systems that contribute knowledge to an Enterprise Knowledge Graph and data and systems that unify the contributing knowledge.  The latter can have a "hub" Ontology as Jim refers to above.  The contributing systems may be able to use different Ontologies as long as they apply them and make them known,  Our past prototyping has introduced a feature in the Enterprise Knowledge Graph environment that manages Ontology via a service that can translate and infer relationships to enable unified knowledge in the Enterprise Knowledge Graph.  We look forward to continued dialog. My work phone is 571-721-3403u for anyone who would like to call me.  Please don't hesitate to contact.

**Comment by neilotte** [comment]
@jimschoening I'm converting this and other items to the Discussion forum where this conversation can continue. Please reach out directly with any concerns. 

For context, please see: https://docs.github.com/en/get-started/using-github/communicating-on-github
